# 1. numpy

## 1-1. 배열 및 명령어

#### 000 numpy 모듈 불러오기 및 버전 확인

- numpy 모듈을 불러오고 버전을 확인해보세요.
- numpy 모듈을 np로 줄여 사용할 수 있도록 작성하세요.

In [1]:
import numpy as np
np.__version__

'1.21.5'

#### 001 타입 확인

- 다음 ```data```변수의 타입을 확인하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = np.array([1, 2, 3])
# 주석을 지우고 이곳에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:red;">Out</span>
```
numpy.ndarray
```

In [2]:
data = np.array([1, 2, 3])
type(data)

numpy.ndarray

#### 002 배열생성

- data 변수를 생성하고 아래의 출력과 같은 값이 나오도록 배열을 생성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
# 주석을 지우고 이곳에 코드를 작성하세요.
data
```
-------
<span style="font-weight:normal; color:red;">Out</span>
```
array([3, 1, 7, 9])
```

In [3]:
data = np.array([3, 1, 7, 9])
data

array([3, 1, 7, 9])

#### 003 배열생성

- ```data1``` 변수를 생성하고 다음의 출력과 같은 값이 나오도록 배열을 생성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
# 주석을 지우고 이곳에 코드를 작성하세요.
data1
```
-------
<span style="font-weight:normal; color:red;">Out</span>
```
array([[3.5 , 4.31, 1.12],
       [9.  , 1.  , 3.  ]])
```

In [4]:
data1 = np.array([[3.5, 4.31, 1.12], [9, 1, 3]])
data1

array([[3.5 , 4.31, 1.12],
       [9.  , 1.  , 3.  ]])

#### 004 차원 확인(ndim)

- ```data1```과 ```data2```가 각각 몇 차원인지 예상해보고 확인하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data1 = np.array([3, 1, 7, 9])
data2 = np.array([[3.5, 4.31, 1.12], [9, 1, 3]])

print(data1) # 주석을 지우고 print 안에 코드를 작성하세요.
print(data2) # 주석을 지우고 print 안에 코드를 작성하세요.
```

In [5]:
data1 = np.array([3, 1, 7, 9])
data2 = np.array([[3.5, 4.31, 1.12], [9, 1, 3]])

print(data1.ndim) # 주석을 지우고 print 안에 코드를 작성하세요.
print(data2.ndim) # 주석을 지우고 print 안에 코드를 작성하세요.

1
2


#### 005 차원 구조 확인(shape)

- ```data1```과 ```data2```의 차원 구조가 어떨지 예상해보고 확인하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data1 = np.array([3, 1, 7, 9])
data2 = np.array([[3.5, 4.31, 1.12], [9, 1, 3]])

print(data1) # 주석을 지우고 print 안에 코드를 작성하세요.
print(data2) # 주석을 지우고 print 안에 코드를 작성하세요.
```

In [6]:
data1 = np.array([3, 1, 7, 9])
data2 = np.array([[3.5, 4.31, 1.12], [9, 1, 3]])

print(data1.shape) # 주석을 지우고 print 안에 코드를 작성하세요.
print(data2.shape) # 주석을 지우고 print 안에 코드를 작성하세요.

(4,)
(2, 3)


#### 006 데이터 타입 확인(dtype)

- ```data1```과 ```data2```의 데이터 타입을 예상해보고 확인하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data1 = np.array([3, 1, 7, 9])
data2 = np.array([[3.5, 4.31, 1.12], [9, 1, 3]])

print(data1) # 주석을 지우고 print 안에 코드를 작성하세요.
print(data2) # 주석을 지우고 print 안에 코드를 작성하세요.
```

In [7]:
data1 = np.array([3, 1, 7, 9])
data2 = np.array([[3.5, 4.31, 1.12], [9, 1, 3]])

print(data1.dtype) # 주석을 지우고 print 안에 코드를 작성하세요.
print(data2.dtype) # 주석을 지우고 print 안에 코드를 작성하세요.

int32
float64


#### 007 데이터 개수 확인(size)

- ```data1```과 ```data2```의 데이터 개수를 예상해보고 확인하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data1 = np.array([3, 1, 7, 9])
data2 = np.array([[3.5, 4.31, 1.12], [9, 1, 3]])

print(data1) # 주석을 지우고 print 안에 코드를 작성하세요.
print(data2) # 주석을 지우고 print 안에 코드를 작성하세요.
```

In [8]:
data1 = np.array([3, 1, 7, 9])
data2 = np.array([[3.5, 4.31, 1.12], [9, 1, 3]])

print(data1.size) # 주석을 지우고 print 안에 코드를 작성하세요.
print(data2.size) # 주석을 지우고 print 안에 코드를 작성하세요.

4
6


#### 008 데이터 타입 지정(dtype)

- ```dtype```이 ```int32```인 ```data```의 타입을 ```float64```로 변경하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = np.array([3, 1, 7, 9])
print(data.dtype)
new_data =  # 주석을 지우고 이곳에 코드를 작성하세요.
print(new_data.dtype)
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
int32
float64
```

In [9]:
data = np.array([3, 1, 7, 9])
print(data.dtype)
new_data = np.array(data, dtype='float')
print(new_data.dtype)

int32
float64


#### 009 행·열의 교환, 전치(Transpose)

- ```data```의 행과 열을 교환하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = np.array([[3., 4., 1., 5.], [9., 1., 3., 2.]])
print(data) # 주석을 지우고 print 안에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
[[3. 9.]
 [4. 1.]
 [1. 3.]
 [5. 2.]]
```

In [55]:
data = np.array([[3., 4., 1., 5.], [9., 1., 3., 2.]])
print(data.T) # 주석을 지우고 print 안에 코드를 작성하세요.

[[3. 9.]
 [4. 1.]
 [1. 3.]
 [5. 2.]]


#### 010 형태 변경(reshape)

- ```data```의 데이터 형태를 보기의 출력값처럼 변경하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = np.array([[3., 4., 1., 5.], [9., 1., 3., 2.]])
print(data) # 주석을 지우고 print 안에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
[[3.]
 [4.]
 [1.]
 [5.]
 [9.]
 [1.]
 [3.]
 [2.]]
```

In [74]:
data = np.array([[3., 4., 1., 5.], [9., 1., 3., 2.]])
print(data.reshape(8, 1)) # 주석을 지우고 print 안에 코드를 작성하세요.

[[3.]
 [4.]
 [1.]
 [5.]
 [9.]
 [1.]
 [3.]
 [2.]]


#### 011 데이터 생성(arange)

- ```arange```를 사용하여 보기와 같은 데이터를 생성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data1 = # 주석을 지우고 이곳에 코드를 작성하세요.
data2 = # 주석을 지우고 이곳에 코드를 작성하세요.
print(data1)
print(data2)
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
[0 1 2 3 4]
[0 2 4 6 8]
```

In [80]:
data1 = np.arange(5)
data2 = np.arange(0, 10, 2)
print(data1)
print(data2)

[0 1 2 3 4]
[0 2 4 6 8]


#### 012 데이터 생성(linspace)

- ```linspace```를 사용하여 보기와 같은 데이터를 생성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data1 = # 주석을 지우고 이곳에 코드를 작성하세요.
data2 = # 주석을 지우고 이곳에 코드를 작성하세요.
print(data1)
print(data2)
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
[10.  5.  0.]
[  0.  25.  50.  75. 100.]
```

In [98]:
data1 = np.linspace(10, 0, 3)
data2 = np.linspace(0, 100, 5)
print(data1)
print(data2)

[10.  5.  0.]
[  0.  25.  50.  75. 100.]


#### 013 데이터 생성(eye)

- ```eye```를 사용하여 보기와 같은 데이터를 생성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data1 = # 주석을 지우고 이곳에 코드를 작성하세요.
data2 = # 주석을 지우고 이곳에 코드를 작성하세요.
print(data1)
print(data2)
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
[[1. 0. 0.]
 [0. 1. 0.]
 [0. 0. 1.]]
[[1 0 0]
 [0 1 0]
 [0 0 1]]
```

In [139]:
data1 = np.eye(3)
data2 = np.eye(3, dtype=np.int16)
print(data1)
print(data2)

[[1. 0. 0.]
 [0. 1. 0.]
 [0. 0. 1.]]
[[1 0 0]
 [0 1 0]
 [0 0 1]]


#### 014 데이터 생성(zeros, ones)

- ```zeros```와 ```ones```를 사용하여 보기와 같은 데이터를 생성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data1 = # 주석을 지우고 이곳에 코드를 작성하세요.
data2 = # 주석을 지우고 이곳에 코드를 작성하세요.
print(data1)
print(data2)
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
[0. 0. 0. 0. 0.]
[[1. 1.]
 [1. 1.]
 [1. 1.]]
```

In [136]:
data1 = np.zeros(5)
data2 = np.ones(shape=(3, 2))
print(data1)
print(data2)

[0. 0. 0. 0. 0.]
[[1. 1.]
 [1. 1.]
 [1. 1.]]


#### 015 데이터 생성(full)

- ```full```을 사용하여 보기와 같은 데이터를 생성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = # 주석을 지우고 이곳에 코드를 작성하세요.
print(data)
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
[[5 5 5 5]
 [5 5 5 5]]
```

In [143]:
data = np.full(shape=(2, 4), fill_value=5)
print(data)

[[5 5 5 5]
 [5 5 5 5]]


#### 016 데이터 생성(randn)

- ```randn```을 사용하여 보기와 같은 정규(가우시안)분포 범위의 난수를 생성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = # 주석을 지우고 이곳에 코드를 작성하세요.
print(data)
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
[[-0.69190273 -0.52112558 -1.72167669 -0.77865483]
 [ 0.83461144  0.06283456 -0.84170318  1.11110502]]
```

In [157]:
data = np.random.randn(2, 4)
print(data)

[[-0.69190273 -0.52112558 -1.72167669 -0.77865483]
 [ 0.83461144  0.06283456 -0.84170318  1.11110502]]


#### 017 기본 연산

- 기본 연산을 활용하여 보기와 같은 값이 나올 수 있도록 코드를 작성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data1 = np.array([[3, 1], [2, 6]])
data2 = np.array([[4, 0], [6, 6]])
print() # 주석을 지우고 print 안에 코드를 작성하세요.
print() # 주석을 지우고 print 안에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
[[-1  1]
 [-4  0]]
[[ 7  1]
 [ 8 12]]
```

In [177]:
data1 = np.array([[3, 1], [2, 6]])
data2 = np.array([[4, 0], [6, 6]])
print(data1 - data2) # 주석을 지우고 print 안에 코드를 작성하세요.
print(data1 + data2) # 주석을 지우고 print 안에 코드를 작성하세요.

[[-1  1]
 [-4  0]]
[[ 7  1]
 [ 8 12]]


#### 018 기본 연산

- 기본 연산을 활용하여 보기와 같은 값이 나올 수 있도록 코드를 작성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data1 = np.array([[3, 1], [2, 6]])
data2 = np.array([[4, 0], [6, 6]])
print() # 주석을 지우고 print 안에 코드를 작성하세요.
print() # 주석을 지우고 print 안에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
[[12  0]
 [12 36]]
[[1.33333333 0.        ]
 [3.         1.        ]]
```

In [181]:
data1 = np.array([[3, 1], [2, 6]])
data2 = np.array([[4, 0], [6, 6]])
print(data1 * data2)
print(data2 / data1)

[[12  0]
 [12 36]]
[[1.33333333 0.        ]
 [3.         1.        ]]


#### 019 기본연산

- 다음은 랜덤한 데이터를 생성하고 데이터를 연산하는 코드입니다.
- 기본연산 함수를 사용하여 요구하는 값을 print하세요.
- 코드는 반드시 print문 안의 중괄호{}에 작성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = np.array(np.random.randn(2, 5) * 10, dtype=int)
display(data)
print(f'전체 합계: {}') # 주석을 지우고 중괄호 안에 코드를 작성하세요.
print(f'행 합계: {}') # 주석을 지우고 중괄호 안에 코드를 작성하세요.
print(f'열 합계: {}') # 주석을 지우고 중괄호 안에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
array([[ -3, -16,   8,   1, -13],
       [  0,  14,   1,  -1, -11]])
전체 합계: -20
행 합계: [ -3  -2   9   0 -24]
열 합계: [-23   3]
```

In [227]:
data = np.array(np.random.randn(2, 5) * 10, dtype=int)
display(data)
print(f'전체 합계: {data.sum()}') # 주석을 지우고 중괄호 안에 코드를 작성하세요.
print(f'행 합계: {data.sum(axis=0)}') # 주석을 지우고 중괄호 안에 코드를 작성하세요.
print(f'열 합계: {data.sum(axis=1)}') # 주석을 지우고 중괄호 안에 코드를 작성하세요.

array([[ -3, -16,   8,   1, -13],
       [  0,  14,   1,  -1, -11]])

전체 합계: -20
행 합계: [ -3  -2   9   0 -24]
열 합계: [-23   3]


#### 020 기본연산

- 다음은 랜덤한 데이터를 생성하고 데이터를 연산하는 코드입니다.
- 기본연산 함수를 사용하여 요구하는 값을 print하세요.
- 코드는 반드시 print문 안의 중괄호{}에 작성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = np.array(np.random.randn(2, 5) * 10, dtype=int)
display(data)
print(f'평균: {}') # 주석을 지우고 중괄호 안에 코드를 작성하세요.
print(f'최대값: {}') # 주석을 지우고 중괄호 안에 코드를 작성하세요.
print(f'최소값: {}') # 주석을 지우고 중괄호 안에 코드를 작성하세요.
print(f'분산: {}') # 주석을 지우고 중괄호 안에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
array([[  7, -12,   0, -11,  -2],
       [  4, -15,  14,   6,   6]])
평균: -0.3
최대값: 14
최소값: -15
분산: 82.61000000000001
```

In [224]:
data = np.array(np.random.randn(2, 5) * 10, dtype=int)
display(data)
print(f'평균: {data.mean()}') # 주석을 지우고 중괄호 안에 코드를 작성하세요.
print(f'최대값: {data.max()}') # 주석을 지우고 중괄호 안에 코드를 작성하세요.
print(f'최소값: {data.min()}') # 주석을 지우고 중괄호 안에 코드를 작성하세요.
print(f'분산: {data.var()}') # 주석을 지우고 중괄호 안에 코드를 작성하세요.

array([[  7, -12,   0, -11,  -2],
       [  4, -15,  14,   6,   6]])

평균: -0.3
최대값: 14
최소값: -15
분산: 82.61000000000001


# 2. pandas

## 2-1. Series

#### 021 pandas 모듈 불러오기 및 버전 확인

- pandas 모듈을 불러오고 버전을 확인해보세요.
- pandas 모듈을 pd 줄여 사용할 수 있도록 작성하세요.

In [5]:
import pandas as pd
pd.__version__

'1.4.4'

#### 022 Series 생성

- list 또는 tuple을 활용하여 보기의 출력값과 같은 Series를 생성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = # 주석을 지우고 이곳에 코드를 작성하세요.
data
```
-------
<span style="font-weight:normal; color:red;">Out</span>
```
0    5
1    3
2    1
3    2
4    4
dtype: int64
```

In [25]:
data = pd.Series([5, 3, 1, 2, 4])
data

0    5
1    3
2    1
3    2
4    4
dtype: int64

#### 023 Series 생성

- numpy.array 활용하여 보기의 출력값과 같은 Series를 생성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = # 주석을 지우고 이곳에 코드를 작성하세요.
data
```
-------
<span style="font-weight:normal; color:red;">Out</span>
```
0    21
1    10
2    36
3    48
4    12
dtype: int32
```

In [21]:
data = pd.Series(np.array([21, 10, 36, 48, 12]))
data

0    21
1    10
2    36
3    48
4    12
dtype: int32

#### 024 Series 생성

- dict를 활용하여 보기의 출력값과 같은 Series를 생성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
temp_dict = # 주석을 지우고 이곳에 코드를 작성하세요.
data = # 주석을 지우고 이곳에 코드를 작성하세요.
data
```
-------
<span style="font-weight:normal; color:red;">Out</span>
```
일    1
이    2
삼    3
사    4
오    5
dtype: int64
```

In [28]:
temp_dict = {'일': 1, '이': 2, '삼': 3, '사': 4, '오': 5}
data = pd.Series(temp_dict)
data

일    1
이    2
삼    3
사    4
오    5
dtype: int64

#### 025 데이터 확인

- 주어진 데이터의 타입을 예상해보고 확인하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data1 = pd.Series(np.array([21, 10, 36, 48, 12]))
data2 = pd.Series(['A', 'B', 'C', 'D', 'E'])
print(data1.dtype)
print(data2.dtype)
data2.dtype
```

In [43]:
data1 = pd.Series(np.array([21, 10, 36, 48, 12]))
data2 = pd.Series(['A', 'B', 'C', 'D', 'E'])
print(data1.dtype)
print(data2.dtype)
data2.dtype

int32
object


dtype('O')

#### 026 데이터 확인

- 다음의 출력문과 같이 주어진 Series의 index와 value를 확인하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = pd.Series({'일': 1, '이': 2, '삼': 3, '사': 4, '오': 5})
print() # 주석을 지우고 print 안에 코드를 작성하세요.
print() # 주석을 지우고 print 안에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
Index(['일', '이', '삼', '사', '오'], dtype='object')
[1 2 3 4 5]
```

In [35]:
data = pd.Series({'일': 1, '이': 2, '삼': 3, '사': 4, '오': 5})
print(data.index) # 주석을 지우고 print 안에 코드를 작성하세요.
print(data.values) # 주석을 지우고 print 안에 코드를 작성하세요.

Index(['일', '이', '삼', '사', '오'], dtype='object')
[1 2 3 4 5]


#### 027 index 수정

- ```data```의 Series 값을 다음의 출력값과 동일하도록 하는 코드를 작성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = pd.Series({'일': 1, '이': 2, '삼': 3, '사': 4, '오': 5})
# 주석을 지우고 이곳에 코드를 작성하세요.
data
```
-------
<span style="font-weight:normal; color:red;">Out</span>
```
A    1
B    2
C    3
D    4
E    5
dtype: int64
```


In [59]:
data = pd.Series({'일': 1, '이': 2, '삼': 3, '사': 4, '오': 5})
#답안1
# data = pd.Series(data.values, index=['A', 'B', 'C', 'D', 'E'])
#답안2
data.index = ['A', 'B', 'C', 'D', 'E']
data

A    1
B    2
C    3
D    4
E    5
dtype: int64

In [61]:
data.values

array([1, 2, 3, 4, 5], dtype=int64)

#### 028 Series 인덱싱

- ```data```의 Series중 출력값과 같이 영어~국사의 데이터에 접근하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = pd.Series({'국어': 77, '영어': 88, '수학': 81, '국사': 63, '과학': 97})
print() # 주석을 지우고 print 안에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
국어    77
영어    88
수학    81
국사    63
dtype: int64
```

In [81]:
data = pd.Series({'국어': 77, '영어': 88, '수학': 81, '국사': 63, '과학': 97})
print(data['국어':'국사'])

국어    77
영어    88
수학    81
국사    63
dtype: int64


#### 029 Series 인덱싱

- 다음의 출력값과 같이 ```data```변수에서 과학 데이터를 조회하세요.
- 반드시 ```at```을 사용하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = pd.Series({'국어': 77, '영어': 88, '수학': 81, '국사': 63, '과학': 97})
print() # 주석을 지우고 print 안에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
97
```

In [82]:
data = pd.Series({'국어': 77, '영어': 88, '수학': 81, '국사': 63, '과학': 97})
print(data.at['과학'])

97


#### 030 Series 연산

- 다음 세 Series 데이터를 연산하여 보기의 출력값과 같도록 코드를 작성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data1 = pd.Series([1, 2, 3, 4, 5])
data2 = pd.Series([10, 20, 30, 40, 50])

print() # 주석을 지우고 print 안에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
0     10
1     40
2     90
3    160
4    250
dtype: int64
```

In [90]:
data1 = pd.Series([1, 2, 3, 4, 5])
data2 = pd.Series([10, 20, 30, 40, 50])

print(data1 * data2)

0     10
1     40
2     90
3    160
4    250
dtype: int64


#### 031 NaN처리

- 다음의 출력값과 같이 변수 ```data```의 NaN을 0으로 처리하는 코드를 작성하세요.
- 반드시 ```fillna```를 사용하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = pd.Series([10, 20, 30, 40, 50]) * pd.Series([1, 2])
# 주석을 지우고 이곳에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:red;">Out</span>
```
0    10.0
1    40.0
2     0.0
3     0.0
4     0.0
dtype: float64
```


In [96]:
data = pd.Series([10, 20, 30, 40, 50]) * pd.Series([1, 2])
data.fillna(0)

0    10.0
1    40.0
2     0.0
3     0.0
4     0.0
dtype: float64

#### 032 Series 데이터 확인

- 다음의 출력값과 같이 Series 데이터의 정보를 확인하는 코드를 작성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = pd.Series([1, 2, 3, 4, 5]) * pd.Series([10, 20, 30, 40, 50])
print('mean:', ) # 주석을 지우고 print 안에 코드를 작성하세요.
print('std:', ) # 주석을 지우고 print 안에 코드를 작성하세요.
print('count:', ) # 주석을 지우고 print 안에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
mean: 110.0
std: 96.69539802906858
count: 5
```

In [123]:
data = pd.Series([1, 2, 3, 4, 5]) * pd.Series([10, 20, 30, 40, 50])
print('mean:', data.mean())
print('std:', data.std())
print('count:', data.count())

mean: 110.0
std: 96.69539802906858
count: 5


#### 033 Series 데이터 확인

- 다음의 출력값과 같이 Series 데이터의 정보를 확인하는 코드를 작성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
data = pd.Series([1, 2, 3, 4, 5]) * pd.Series([10, 20, 30, 40, 50])
print() # 주석을 지우고 print 안에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:orange;">Print</span>
```
count      5.000000
mean     110.000000
std       96.695398
min       10.000000
25%       40.000000
50%       90.000000
75%      160.000000
max      250.000000
dtype: float64
```

In [128]:
data = pd.Series([1, 2, 3, 4, 5]) * pd.Series([10, 20, 30, 40, 50])
print(data.describe())

count      5.000000
mean     110.000000
std       96.695398
min       10.000000
25%       40.000000
50%       90.000000
75%      160.000000
max      250.000000
dtype: float64


## 2-2. DataFrame

#### 034 DataFrame 생성

- list를 활용해 보기의 출력값과 같은 DataFrame을 생성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = # 주석을 지우고 이곳에 코드를 작성하세요.
df
```
-------
<span style="font-weight:normal; color:red;">Out</span>

||0|1|2|3|
|--|--|--|--|--|
|**0**|	0|	7|	8|	1|
|**1**|	1|	4|	7|	3|


In [151]:
df = pd.DataFrame([[0, 7, 8, 1],[1, 4, 7, 3]])
df

0  1  2  3
0  0  7  8  1
1  1  4  7  3

#### 035 DataFrame 생성

- dict를 활용해 보기의 출력값과 같은 DataFrame을 생성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = # 주석을 지우고 이곳에 코드를 작성하세요.
df
```
-------
<span style="font-weight:normal; color:red;">Out</span>

||학년|수학|영어|
|--|--|--|--|
|**0**|	0|	33|	98|
|**1**|	1|	79|	67|
|**2**|	3|	81|	31|
|**3**|	2|	50|	49|

In [152]:
df = pd.DataFrame({'학년':[1,1,3,2], '수학':[33, 79, 81, 50], '영어':[98, 67, 31, 49]})
df

학년  수학  영어
0   1  33  98
1   1  79  67
2   3  81  31
3   2  50  49

#### 036 DataFrame 생성

- 아래의 출력값과 같이 index와 column이 출력 될 수 있도록 DataFrame을 생성하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = # 주석을 지우고 이곳에 DataFrame을 생성하세요.
df
```
-------
<span style="font-weight:normal; color:red;">Out</span>

||남|여|
|--|--|--|
|**김xx**|1|0|
|**이xx**|0|1|
|**박xx**|0|1|
|**최xx**|1|0|

In [48]:
df = pd.DataFrame([[1, 0], [0, 1], [0, 1], [1, 0]],
                  index=['김xx', '이xx', '박xx', '최xx'],
                  columns=['남', '여'])
df

남  여
김xx  1  0
이xx  0  1
박xx  0  1
최xx  1  0

#### 037 index, column 변경

- 다음 데이터 프레임의 index와 column을 보기와 같이 변경하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = pd.DataFrame({'a':[37.58945628, 37.60935675, 37.59167275, 37.60935675, 37.60963055],
                   'b':[127.0168496, 127.0530356, 127.0122079, 127.0530356, 127.0530544]})
# 주석을 지우고 이곳에 코드를 작성하세요.
df
```
-------
<span style="font-weight:normal; color:red;">Out</span>

||위도|경도|
|--|--|--|
|**A**|37.58945628|127.0168496|
|**B**|37.60935675|127.0530356|
|**C**|37.59167275|127.0122079|
|**D**|37.60935675|127.0530356|
|**E**|37.60963055|127.0530544|


In [55]:
df = pd.DataFrame({'a':[37.58945628, 37.60935675, 37.59167275, 37.60935675, 37.60963055],
                   'b':[127.0168496, 127.0530356, 127.0122079, 127.0530356, 127.0530544]})
df.index = ['A', 'B', 'C', 'D', 'E']
df.columns = ['위도', '경도']
df

위도          경도
A  37.589456  127.016850
B  37.609357  127.053036
C  37.591673  127.012208
D  37.609357  127.053036
E  37.609631  127.053054

#### 038 데이터 조회

- 다음 ```df``` 데이터의 1번 행의 값을 조회하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = pd.DataFrame({'구분':['일자리','창업','창업','창업','소공인'],
                   '시설명':['일자리 플러스 센터','벤처창업지원센터','1인 창조기업 비즈니스센터','시니어 기술창업센터','성북구 패션봉제지원센터(소공인특화지원센터)'],
                   '행정동':['삼선동','장위2동','성북동','장위2동','장위2동']})
# 주석을 지우고 이곳에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:red;">Out</span>
```
구분           창업
시설명    벤처창업지원센터
행정동        장위2동
Name: 1, dtype: object
```

In [73]:
df = pd.DataFrame({'구분':['일자리','창업','창업','창업','소공인'],
                   '시설명':['일자리 플러스 센터','벤처창업지원센터','1인 창조기업 비즈니스센터','시니어 기술창업센터','성북구 패션봉제지원센터(소공인특화지원센터)'],
                   '행정동':['삼선동','장위2동','성북동','장위2동','장위2동']})
df.loc[1]

구분           창업
시설명    벤처창업지원센터
행정동        장위2동
Name: 1, dtype: object

#### 039 데이터 조회

- 다음 ```df``` 데이터의 1~3번을 포함한 행의 값을 조회하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = pd.DataFrame({'구분':['일자리','창업','창업','창업','소공인'],
                   '시설명':['일자리 플러스 센터','벤처창업지원센터','1인 창조기업 비즈니스센터','시니어 기술창업센터','성북구 패션봉제지원센터(소공인특화지원센터)'],
                   '행정동':['삼선동','장위2동','성북동','장위2동','장위2동']})

print()# 주석을 지우고 print 안에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:orange;">Print</span>

```
    구분                      시설명   행정동
1   창업                 벤처창업지원센터  장위2동
2   창업           1인 창조기업 비즈니스센터   성북동
3   창업               시니어 기술창업센터  장위2동
4  소공인  성북구 패션봉제지원센터(소공인특화지원센터)  장위2동
```

In [77]:
df = pd.DataFrame({'구분':['일자리','창업','창업','창업','소공인'],
                   '시설명':['일자리 플러스 센터','벤처창업지원센터','1인 창조기업 비즈니스센터','시니어 기술창업센터','성북구 패션봉제지원센터(소공인특화지원센터)'],
                   '행정동':['삼선동','장위2동','성북동','장위2동','장위2동']})

print(df.loc[1:4])

    구분                      시설명   행정동
1   창업                 벤처창업지원센터  장위2동
2   창업           1인 창조기업 비즈니스센터   성북동
3   창업               시니어 기술창업센터  장위2동
4  소공인  성북구 패션봉제지원센터(소공인특화지원센터)  장위2동


#### 040 데이터 조회

- 다음의 출력값 처럼 ```df``` 의 데이터를 조회하세요.
- 반드시 인덱스와 컬럼을 동시에 사용하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = pd.DataFrame({'구분':['일자리','창업','창업','창업','소공인'],
                   '시설명':['일자리 플러스 센터','벤처창업지원센터','1인 창조기업 비즈니스센터','시니어 기술창업센터','성북구 패션봉제지원센터(소공인특화지원센터)'],
                   '행정동':['삼선동','장위2동','성북동','장위2동','장위2동']})

# 주석을 지우고 이곳에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:red;">Out</span>

||구분|행정동|
|--|--|--|
|2|창업|성북동|
|3|창업|장위2동|
|4|소공인|장위2동|


In [101]:
df = pd.DataFrame({'구분':['일자리','창업','창업','창업','소공인'],
                   '시설명':['일자리 플러스 센터','벤처창업지원센터','1인 창조기업 비즈니스센터','시니어 기술창업센터','성북구 패션봉제지원센터(소공인특화지원센터)'],
                   '행정동':['삼선동','장위2동','성북동','장위2동','장위2동']})
df.loc[2:5, ['구분','행정동']]

구분   행정동
2   창업   성북동
3   창업  장위2동
4  소공인  장위2동

#### 041 데이터 조회

- 다음 출력값과 같이 위도가 37.6 보다 높은 데이터만 조회하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = pd.DataFrame({'위도':[37.58945628, 37.60935675, 37.59167275, 37.60935675, 37.60963055],
                   '경도':[127.0168496, 127.0530356, 127.0122079, 127.0530356, 127.0530544]})

# 주석을 지우고 이곳에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:red;">Out</span>

||위도|경도|
|--|--|--|
|**1**|37.609357|127.053036|
|**3**|37.609357|127.053036|
|**4**|37.609631|127.053054|


In [107]:
df = pd.DataFrame({'위도':[37.58945628, 37.60935675, 37.59167275, 37.60935675, 37.60963055],
                   '경도':[127.0168496, 127.0530356, 127.0122079, 127.0530356, 127.0530544]})
df[df['위도']>37.6]

위도          경도
1  37.609357  127.053036
3  37.609357  127.053036
4  37.609631  127.053054

#### 042 데이터 합치기

- 다음의 두 데이터를 index 기준으로 합치세요

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df1 = pd.DataFrame({'위도':[37.58945628, 37.60935675, 37.59167275, 37.60935675, 37.60963055],
                    '경도':[127.0168496, 127.0530356, 127.0122079, 127.0530356, 127.0530544]})
df2 = pd.DataFrame({'구분':['일자리','창업','창업','창업','소공인'],
                    '시설명':['일자리 플러스 센터','벤처창업지원센터','1인 창조기업 비즈니스센터','시니어 기술창업센터','성북구 패션봉제지원센터(소공인특화지원센터)'],
                    '행정동':['삼선동','장위2동','성북동','장위2동','장위2동']})

# 주석을 지우고 이곳에 코드를 작성하세요.
```
-------
<span style="font-weight:normal; color:red;">Out</span>

||위도|경도|구분|시설명|행정동|
|--|--|--|--|--|--|
|**0**|37.589456|127.016850|일자리|일자리 플러스 센터|삼선동|
|**1**|37.609357|127.053036|창업|벤처창업지원센터	|장위2동|
|**2**|37.591673|127.012208|창업|1인 창조기업 비즈니스센터|성북동|
|**3**|37.609357|127.053036|창업|시니어 기술창업센터|장위2동|
|**4**|37.609631|127.053054|소공인|성북구 패션봉제지원센터(소공인특화지원센터)|장위2동|

In [151]:
df1 = pd.DataFrame({'위도':[37.58945628, 37.60935675, 37.59167275, 37.60935675, 37.60963055],
                    '경도':[127.0168496, 127.0530356, 127.0122079, 127.0530356, 127.0530544]})
df2 = pd.DataFrame({'구분':['일자리','창업','창업','창업','소공인'],
                    '시설명':['일자리 플러스 센터','벤처창업지원센터','1인 창조기업 비즈니스센터','시니어 기술창업센터','성북구 패션봉제지원센터(소공인특화지원센터)'],
                    '행정동':['삼선동','장위2동','성북동','장위2동','장위2동']})
pd.merge(left=df1, right=df2, left_index=True, right_index=True, how='inner')

위도          경도   구분                      시설명   행정동
0  37.589456  127.016850  일자리               일자리 플러스 센터   삼선동
1  37.609357  127.053036   창업                 벤처창업지원센터  장위2동
2  37.591673  127.012208   창업           1인 창조기업 비즈니스센터   성북동
3  37.609357  127.053036   창업               시니어 기술창업센터  장위2동
4  37.609631  127.053054  소공인  성북구 패션봉제지원센터(소공인특화지원센터)  장위2동

#### 043 데이터프레임 연산

- 두 데이터프레임의 연산 결과를 예상해보고 확인하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df1 = pd.DataFrame([[12, 6, 3], [4, 12, 9]])
df2 = pd.DataFrame([[3, 6], [1, 6]])

df1 / df2
```

In [160]:
df1 = pd.DataFrame([[12, 6, 3], [4, 12, 9]])
df2 = pd.DataFrame([[3, 6], [1, 6]])

df1 / df2

0    1   2
0  4.0  1.0 NaN
1  4.0  2.0 NaN

#### 044 csv 불러오기

- 다음의 링크로 들어가 공공데이터를 다운받으세요.[[https://www.data.go.kr/data/15040314/fileData.do#/tab-layer-file]](https://www.data.go.kr/data/15040314/fileData.do#/tab-layer-file)
- 다운 받은 데이터를 현재 작업경로로 옮기세요.
- csv 데이터를 불러와 df 변수에 저장하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = # 주석을 지우고 이곳에 코드를 작성하세요.
df.head()
```

In [198]:
df = pd.read_csv('서울특별시 성북구_일자리지원관련센터_20210101.csv', encoding='cp949')
df.head()

구분                      시설명                                     주소  \
0  일자리               일자리 플러스 센터             서울특별시 성북구 보문로 168, 성북구청 2층   
1   창업                 벤처창업지원센터                      서울특별시 성북구 화랑로 211   
2   창업           1인 창조기업 비즈니스센터           서울특별시 성북구 동소문로 63, 드림트리빌딩 6층   
3   창업               시니어 기술창업센터  서울특별시 성북구 화랑로 211, 성북구벤처창업지원센터 내 지하1층   
4  소공인  성북구 패션봉제지원센터(소공인특화지원센터)                   서울특별시 성북구 돌곶이로25길 73   

          위도          경도                                 지원내용_등          전화번호  \
0  37.589456  127.016850          맞춤형 취업상담, 알선 및 구인기업 발굴 등록관리 등  02-2241-3947   
1  37.609357  127.053036               벤처기업의 창업 공간 및 보육 프로그램 제공   02-941-6639   
2  37.591673  127.012208            1인 창조 산업 창업자 및 예비 창업자 지원 육성  02-2241-3984   
3  37.609357  127.053036                    시니어(40세이상) 기술 창업 지원   02-941-7257   
4  37.609631  127.053054  봉제산업 소공인 경쟁력 제고를 위한 교육 제공 및 봉제장비 임대지원   02-919-9257   

            위탁기관   행정동   비고  
0            NaN   삼선동  NaN  
1          한성대학교  장위2동  NaN  
2        성북구청 직영   성북동  NaN  
3          한성대학교  장위2동  NaN  
4  (사)서울패션섬유봉제협회  장위2동  NaN

#### 045 DataFrame 통계

- 다음의 링크로 들어가 공공데이터를 다운받으세요.[[공공데이터]](https://www.data.go.kr/data/15105026/fileData.do)
- 다운 받은 데이터를 현재 작업경로로 옮기세요.
- ```경기도 여주시_보호수_20220803.csv``` 파일을 불러와 ```df``` 변수에 저장하세요.
- ```df```의 통계 요약을 출력하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = pd.read_csv('경기도 여주시_보호수_20220803.csv', encoding='cp949')
# 주석을 지우고 이곳에 코드를 작성하세요.
```

In [253]:
df = pd.read_csv('경기도 여주시_보호수_20220803.csv', encoding='cp949')
df.describe()

그루수        나무나이        나무높이      가슴높이둘레      나무갓지름          위도  \
count  209.000000  209.000000  209.000000  209.000000  68.000000  209.000000   
mean     1.028708  251.837321   15.258373  182.781866   1.335441   37.326528   
std      0.238462  131.792058    5.214646  209.697709   0.613508    0.083628   
min      1.000000  100.000000    6.000000    0.330000   0.330000   36.593942   
25%      1.000000  150.000000   12.000000    1.700000   0.900000   37.289290   
50%      1.000000  200.000000   16.000000  118.000000   1.220000   37.345050   
75%      1.000000  300.000000   20.000000  252.000000   1.700000   37.379271   
max      3.000000  700.000000   25.000000  982.000000   3.700000   37.423440   

               경도  
count  209.000000  
mean   127.588454  
std      0.084569  
min    127.033899  
25%    127.537001  
50%    127.596407  
75%    127.653692  
max    127.720102

#### 046 DataFrame 통계

- ```경기도 여주시_보호수_20220803.csv``` 파일을 불러와 ```df``` 변수에 저장하세요.
- ```df```의 나무나이, 나무높이, 가슴높이둘레, 나무갓지름의 상관계수를 출력하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = pd.read_csv('경기도 여주시_보호수_20220803.csv', encoding='cp949')
# 주석을 지우고 이곳에 코드를 작성하세요.
```

In [252]:
df = pd.read_csv('경기도 여주시_보호수_20220803.csv', encoding='cp949')
df[['나무나이', '나무높이', '가슴높이둘레', '나무갓지름']].corr()

나무나이      나무높이    가슴높이둘레     나무갓지름
나무나이    1.000000  0.368000  0.196830  0.551295
나무높이    0.368000  1.000000  0.285285  0.681745
가슴높이둘레  0.196830  0.285285  1.000000  0.840712
나무갓지름   0.551295  0.681745  0.840712  1.000000

#### 047 DataFrame 통계

- ```경기도 여주시_보호수_20220803.csv``` 파일을 불러와 ```df``` 변수에 저장하세요.
- ```df```의 그루수, 나무나이 컬럼의 합을 출력하세요

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = pd.read_csv('경기도 여주시_보호수_20220803.csv', encoding='cp949')
# 주석을 지우고 이곳에 코드를 작성하세요.
```

In [256]:
df = pd.read_csv('경기도 여주시_보호수_20220803.csv', encoding='cp949')
df[['그루수', '나무나이']].sum()

그루수       215
나무나이    52634
dtype: int64

#### 048 DataFrame 통계

- ```경기도 여주시_보호수_20220803.csv``` 파일을 불러와 ```df``` 변수에 저장하세요.
- ```df```를 나무종류 순서로 정렬하세요.
- 내림차순으로 정렬하세요.
- 변경된 데이터를 원본 데이터에 덮어 쓸 수 있는 파라미터를 사용하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = pd.read_csv('경기도 여주시_보호수_20220803.csv', encoding='cp949')
# 주석을 지우고 이곳에 코드를 작성하세요.
df.head()
```

In [283]:
df = pd.read_csv('경기도 여주시_보호수_20220803.csv', encoding='cp949')
df.sort_values(by='나무종류', axis=0, ascending=False, inplace=True)
df.head()

관리기관명   지정번호     보호수지정일자 보호수유형명                      학명 나무종류  그루수  \
56   경기도 여주시청  여주-65  1982-10-15     노목  Juniperus chinensis L.  향나무    1   
81   경기도 여주시청  여주-65  1982-10-15     노목                     NaN  향나무    1   
121  경기도 여주시청  여주-18  1982-10-15     노목                     NaN  향나무    1   
150  경기도 여주시청  여주-67  1982-10-15     노목                     NaN  향나무    1   
73   경기도 여주시청  여주-74  1982-10-15     노목                     NaN  향나무    1   

     나무나이  나무높이  가슴높이둘레  나무갓지름 나무품격명 나무지목명                소재지지번주소         위도  \
56    500     7  220.00   0.58  마을나무  종교용지        경기도 여주시 천송동 282  37.297760   
81    500     7    0.58    NaN  마을나무  종교용지        경기도 여주시 천송동 282  37.297760   
121   160     7    0.50    NaN  마을나무     대  경기도 여주시 능서면 번도리 321-4  37.306140   
150   120     7   52.00    NaN  마을나무     대     경기도 여주시 천송동 510-37  37.301251   
73    150     8    1.50    NaN  마을나무     전      경기도 여주시 현암동 320-2  37.314760   

             경도     데이터기준일자  
56   127.661310  2022-08-03  
81   127.661310  2021-08-30  
121  127.576640  2021-08-30  
150  127.653692  2020-07-17  
73   127.636750  2021-08-30

#### 049 DataFrame 통계

- ```경기도 여주시_보호수_20220803.csv``` 파일을 불러와 ```df``` 변수에 저장하세요.
- ```df```를 컬럼의 이름 순서로 정렬하세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = pd.read_csv('경기도 여주시_보호수_20220803.csv', encoding='cp949')
# 주석을 지우고 이곳에 코드를 작성하세요.
```

In [284]:
df = pd.read_csv('경기도 여주시_보호수_20220803.csv', encoding='cp949')
df.sort_index(axis=1)

가슴높이둘레          경도     관리기관명  그루수  나무갓지름  나무나이  나무높이  나무종류 나무지목명 나무품격명  \
0     685.0  127.680500  경기도 여주시청    1   2.08   300    18  느티나무    하천  마을나무   
1     320.0  127.628410  경기도 여주시청    1   1.16   200    20  은행나무     대  마을나무   
2     179.0  127.628160  경기도 여주시청    1   0.44   100     6   향나무     대  마을나무   
3     124.0  127.628070  경기도 여주시청    1   0.33   150     6   향나무     대  마을나무   
4     358.0  127.666180  경기도 여주시청    1   1.00   250     7   향나무    도로  마을나무   
..      ...         ...       ...  ...    ...   ...   ...   ...   ...   ...   
204   100.0  127.662394  경기도 여주시청    1    NaN   250     7   향나무    도로  마을나무   
205    33.0  127.628319  경기도 여주시청    1    NaN   150     6   향나무     대  마을나무   
206    44.0  127.628319  경기도 여주시청    1    NaN   100     6   향나무     대  마을나무   
207   116.0  127.628319  경기도 여주시청    1    NaN   200    20  은행나무     대  마을나무   
208   208.0  127.033899  경기도 여주시청    1    NaN   300    18  느티나무    하천  마을나무   

        데이터기준일자 보호수유형명     보호수지정일자           소재지지번주소         위도  지정번호  \
0    2022-08-03     노목  1982-10-15   경기도 여주시 우만동 233  37.255870  여주-1   
1    2022-08-03     노목  1982-10-15  경기도 여주시 교동 261-1  37.289250  여주-2   
2    2022-08-03     노목  1982-10-15  경기도 여주시 교동 261-1  37.289240  여주-4   
3    2022-08-03     노목  1982-10-15  경기도 여주시 교동 261-1  37.289290  여주-5   
4    2022-08-03     노목  1982-10-15   경기도 여주시 멱곡동 492  37.255370  여주-6   
..          ...    ...         ...               ...        ...   ...   
204  2020-07-17     노목  1982-10-15   경기도 여주시 멱곡동 492  37.248255  여주-6   
205  2020-07-17     노목  1982-10-15  경기도 여주시 교동 261-1  37.289136  여주-5   
206  2020-07-17     노목  1982-10-15  경기도 여주시 교동 261-1  37.289136  여주-4   
207  2020-07-17     노목  1982-10-15  경기도 여주시 교동 261-1  37.289136  여주-2   
208  2020-07-17     노목  1982-10-15   경기도 여주시 우만동 233  37.285344  여주-1   

                                  학명  
0    Zelkova serrata (Thunb.) Makino  
1                   Ginkgo biloba L.  
2             Juniperus chinensis L.  
3             Juniperus chinensis L.  
4             Juniperus chinensis L.  
..                               ...  
204                              NaN  
205                              NaN  
206                              NaN  
207                              NaN  
208                              NaN  

[209 rows x 17 columns]

#### 050 결측치 처리

- ```경기도 여주시_보호수_20220803.csv``` 파일을 불러와 ```df``` 변수에 저장하세요.
- ```df```의 컬럼 중 나무갓지름의 빈 데이터가 포함된 곳에 나무갓지름의 평균값을 채워넣으세요.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
# 주석을 지우고 이곳에 코드를 작성하세요.
```

In [360]:
df = pd.read_csv('경기도 여주시_보호수_20220803.csv', encoding='cp949')
df['나무갓지름'].fillna(df['나무갓지름'].mean(), inplace=True)
df

관리기관명  지정번호     보호수지정일자 보호수유형명                               학명  나무종류  \
0    경기도 여주시청  여주-1  1982-10-15     노목  Zelkova serrata (Thunb.) Makino  느티나무   
1    경기도 여주시청  여주-2  1982-10-15     노목                 Ginkgo biloba L.  은행나무   
2    경기도 여주시청  여주-4  1982-10-15     노목           Juniperus chinensis L.   향나무   
3    경기도 여주시청  여주-5  1982-10-15     노목           Juniperus chinensis L.   향나무   
4    경기도 여주시청  여주-6  1982-10-15     노목           Juniperus chinensis L.   향나무   
..        ...   ...         ...    ...                              ...   ...   
204  경기도 여주시청  여주-6  1982-10-15     노목                              NaN   향나무   
205  경기도 여주시청  여주-5  1982-10-15     노목                              NaN   향나무   
206  경기도 여주시청  여주-4  1982-10-15     노목                              NaN   향나무   
207  경기도 여주시청  여주-2  1982-10-15     노목                              NaN  은행나무   
208  경기도 여주시청  여주-1  1982-10-15     노목                              NaN  느티나무   

     그루수  나무나이  나무높이  가슴높이둘레     나무갓지름 나무품격명 나무지목명           소재지지번주소  \
0      1   300    18   685.0  2.080000  마을나무    하천   경기도 여주시 우만동 233   
1      1   200    20   320.0  1.160000  마을나무     대  경기도 여주시 교동 261-1   
2      1   100     6   179.0  0.440000  마을나무     대  경기도 여주시 교동 261-1   
3      1   150     6   124.0  0.330000  마을나무     대  경기도 여주시 교동 261-1   
4      1   250     7   358.0  1.000000  마을나무    도로   경기도 여주시 멱곡동 492   
..   ...   ...   ...     ...       ...   ...   ...               ...   
204    1   250     7   100.0  1.335441  마을나무    도로   경기도 여주시 멱곡동 492   
205    1   150     6    33.0  1.335441  마을나무     대  경기도 여주시 교동 261-1   
206    1   100     6    44.0  1.335441  마을나무     대  경기도 여주시 교동 261-1   
207    1   200    20   116.0  1.335441  마을나무     대  경기도 여주시 교동 261-1   
208    1   300    18   208.0  1.335441  마을나무    하천   경기도 여주시 우만동 233   

            위도          경도     데이터기준일자  
0    37.255870  127.680500  2022-08-03  
1    37.289250  127.628410  2022-08-03  
2    37.289240  127.628160  2022-08-03  
3    37.289290  127.628070  2022-08-03  
4    37.255370  127.666180  2022-08-03  
..         ...         ...         ...  
204  37.248255  127.662394  2020-07-17  
205  37.289136  127.628319  2020-07-17  
206  37.289136  127.628319  2020-07-17  
207  37.289136  127.628319  2020-07-17  
208  37.285344  127.033899  2020-07-17  

[209 rows x 17 columns]

#### 051 데이터 조회 및 변경

- ```경기도 여주시_보호수_20220803.csv``` 파일을 불러와 ```df``` 변수에 저장하세요.
- ```df```의 컬럼 중 학명의 빈 데이터가 포함된 알맞는 데이터를 채워 넣으세요.


- ```느티나무```의 학명은 ```Zelkova serrata (Thunb.) Makino``` 입니다.
- ```물푸레나무```의 학명은 ```Fraxinus rhynchophylla``` 입니다.
- ```아까시나무```의 학명은 ```Robinia pseudoacacia``` 입니다.
- ```오리나무```의 학명은 ```Alnus japonica (Thunb.) Steudel``` 입니다.
- ```은행나무```의 학명은 ```Ginkgo biloba L.``` 입니다.
- ```향나무```의 학명은 ```Juniperus chinensis``` 입니다.

-------
<span style="font-weight:normal; color:#303F9F;">In</span>
```python
df = pd.read_csv('경기도 여주시_보호수_20220803.csv', encoding='cp949')
# 주석을 지우고 이곳에 코드를 작성하세요.
df
```

In [359]:
df = pd.read_csv('경기도 여주시_보호수_20220803.csv', encoding='cp949')
df.loc[df['나무종류']=='느티나무', '학명'] = 'Zelkova serrata (Thunb.) Makino'
df.loc[df['나무종류']=='물푸레나무', '학명'] = 'Fraxinus rhynchophylla'
df.loc[df['나무종류']=='아까시나무', '학명'] = 'Robinia pseudoacacia'
df.loc[df['나무종류']=='오리나무', '학명'] = 'Alnus japonica (Thunb.) Steudel'
df.loc[df['나무종류']=='은행나무', '학명'] = 'Ginkgo biloba L.'
df.loc[df['나무종류']=='향나무', '학명'] = 'Juniperus chinensis'
df

관리기관명  지정번호     보호수지정일자 보호수유형명                               학명  나무종류  \
0    경기도 여주시청  여주-1  1982-10-15     노목  Zelkova serrata (Thunb.) Makino  느티나무   
1    경기도 여주시청  여주-2  1982-10-15     노목                 Ginkgo biloba L.  은행나무   
2    경기도 여주시청  여주-4  1982-10-15     노목              Juniperus chinensis   향나무   
3    경기도 여주시청  여주-5  1982-10-15     노목              Juniperus chinensis   향나무   
4    경기도 여주시청  여주-6  1982-10-15     노목              Juniperus chinensis   향나무   
..        ...   ...         ...    ...                              ...   ...   
204  경기도 여주시청  여주-6  1982-10-15     노목              Juniperus chinensis   향나무   
205  경기도 여주시청  여주-5  1982-10-15     노목              Juniperus chinensis   향나무   
206  경기도 여주시청  여주-4  1982-10-15     노목              Juniperus chinensis   향나무   
207  경기도 여주시청  여주-2  1982-10-15     노목                 Ginkgo biloba L.  은행나무   
208  경기도 여주시청  여주-1  1982-10-15     노목  Zelkova serrata (Thunb.) Makino  느티나무   

     그루수  나무나이  나무높이  가슴높이둘레  나무갓지름 나무품격명 나무지목명           소재지지번주소         위도  \
0      1   300    18   685.0   2.08  마을나무    하천   경기도 여주시 우만동 233  37.255870   
1      1   200    20   320.0   1.16  마을나무     대  경기도 여주시 교동 261-1  37.289250   
2      1   100     6   179.0   0.44  마을나무     대  경기도 여주시 교동 261-1  37.289240   
3      1   150     6   124.0   0.33  마을나무     대  경기도 여주시 교동 261-1  37.289290   
4      1   250     7   358.0   1.00  마을나무    도로   경기도 여주시 멱곡동 492  37.255370   
..   ...   ...   ...     ...    ...   ...   ...               ...        ...   
204    1   250     7   100.0    NaN  마을나무    도로   경기도 여주시 멱곡동 492  37.248255   
205    1   150     6    33.0    NaN  마을나무     대  경기도 여주시 교동 261-1  37.289136   
206    1   100     6    44.0    NaN  마을나무     대  경기도 여주시 교동 261-1  37.289136   
207    1   200    20   116.0    NaN  마을나무     대  경기도 여주시 교동 261-1  37.289136   
208    1   300    18   208.0    NaN  마을나무    하천   경기도 여주시 우만동 233  37.285344   

             경도     데이터기준일자  
0    127.680500  2022-08-03  
1    127.628410  2022-08-03  
2    127.628160  2022-08-03  
3    127.628070  2022-08-03  
4    127.666180  2022-08-03  
..          ...         ...  
204  127.662394  2020-07-17  
205  127.628319  2020-07-17  
206  127.628319  2020-07-17  
207  127.628319  2020-07-17  
208  127.033899  2020-07-17  

[209 rows x 17 columns]